tools/analyzer_plugin/test/unit/util/prop_forwarding_test.dart

```diff
@@ -0,0 +1,1009 @@
+import 'package:analyzer/dart/element/element.dart';
+import 'package:over_react_analyzer_plugin/src/diagnostic_contributor.dart';
+import 'package:over_react_analyzer_plugin/src/util/prop_forwarding/forwarded_props.dart';
+import 'package:test/test.dart';
+
+import '../../util/shared_analysis_context.dart';
+import 'prop_declaration/util.dart';
+
+void main() {
+  group('prop_forwarding:', () {
+    final sharedContext = SharedAnalysisContext.overReact;
+
+    setUpAll(sharedContext.warmUpAnalysis);
+
+    group('computeForwardedProps correctly computes forwarded props for various cases:', () {
+      const sharedSourcePrefix = /*language=dart*/ r'''
+          // @dart=2.12
+          import 'package:over_react/over_react.dart';
+          
+          part '{{PART_PATH}}';
+          
+          // [1] Create static meta variables that we only support for legacy props.
+          //     We want to test referencing them, and we'll do so with these mixins
+          //     to be able to reuse this test setup.
+          // [2] We can skip most all the analyzer ignore comments you'd usually need for this boilerplate,
+          //     since these tests generate code first before analyzing.
+          
+          mixin AProps on UiProps {
+            static const meta = $AProps.meta; // [1]
+                        
+            String? a;
+          }
+          mixin BProps on UiProps {
+            static const meta = $BProps.meta; // [1]
+            
+            String? b;
+          }
+          mixin CProps on UiProps {
+            static const meta = $CProps.meta; // [1]
+            
+            String? c;
+          }
+          
+          class HasABCProps = UiProps with AProps, BProps, CProps;
+          
+          mixin UnrelatedProps on UiProps {}
+          
+          late UiFactory<UiProps> NotCare;
+          
+          // [2]
+          @AbstractProps()
+          abstract class _$LegacyUnrelatedProps extends UiProps {}
+          abstract class LegacyUnrelatedProps extends _$LegacyUnrelatedProps with _$LegacyUnrelatedPropsAccessorsMixin {
+            static const PropsMeta meta = _$metaForLegacyUnrelatedProps;
+          }
+          
+          late bool condition;
+      ''';
+
+      Future<ResolvedUnitResult> sharedResolveSource(String source) =>
+          resolveFileAndGeneratedPart(sharedContext, '$sharedSourcePrefix\n$source');
+
+      group('adding `props` without filtering:', () {
+        for (final methodName in ['addAll', 'addProps']) {
+          group('via $methodName:', () {
+            test('', () async {
+              final result = await sharedResolveSource(/*language=dart*/ r'''
+                  UiFactory<HasABCProps> HasABC = uiFunction((props) {
+                    return (NotCare()
+                      ..METHOD_NAME(props)
+                    )();
+                  }, _$HasABCConfig);
+              '''
+                  .replaceAll('METHOD_NAME', methodName));
```

Comment:
Went with replaceAll 
- so I could use raw strings and not have to escape the $s in the boilerplate lol
- so that the `language=dart` language injection wouldn't get messed up by string interpolations